ls1intum/Artemis
# Always display the latest five exercises in "Past Exercises"

### Checklist

### Motivation and Context

Currently a lot of space is wasted if there are no active exercises (i.e. no exercises with an assessment due date within the last seven days):
![grafik](https://user-images.githubusercontent.com/72132281/112679839-ed132880-8e6c-11eb-9215-2a795dd70477.png)


### Description

We now always display the five latest exercises in the "Past Exercises" category. If there are no exercises in the course the old "No active exercises to display" is still displayed. The category title changed to "Past Exercises (N of M)" where N is the number of displayed exercises and M is the total number of past exercises.

### Steps for Testing

1. Navigate to the Course Management
2. Check every course with past exercises has five exercises displayed (if there are that many past exercises)
3. Check that exercises with the most recent assessment due date (or due date if there is no assessment due date) are displayed on top
4. Make sure that the statistics the exercises display in each category (participations, assessment progress, average score) behave like on develop

### Screenshots

UI wise nothing much has changed, but more courses will have something to display now:
![grafik](https://user-images.githubusercontent.com/72132281/112677660-53e31280-8e6a-11eb-8f83-fae6cdc67267.png)
In German:
![grafik](https://user-images.githubusercontent.com/72132281/112677815-77a65880-8e6a-11eb-96d0-b1d34070fb7d.png)
Less than five past exercises:
![grafik](https://user-images.githubusercontent.com/72132281/112677879-8a209200-8e6a-11eb-93dd-3c47d9b8c806.png)



### Test Coverage

Tests have been adjusted to cover all new methods and functions.
